File: pathplannerlib/src/test/native/cpp/pathplanner/lib/PathConstraintsTest.cpp
#include <gtest/gtest.h>
#include "pathplanner/lib/PathConstraints.h"

using namespace pathplanner;

TEST(PathConstraintsTest, TestConstructor) {
	PathConstraints constraints(4_mps, 3_mps_sq);

	EXPECT_EQ(4_mps, constraints.maxVelocity);
	EXPECT_EQ(3_mps_sq, constraints.maxAcceleration);
}

File: pathplannerlib/src/test/java/com/pathplanner/lib/PathConstraintsTest.java
package com.pathplanner.lib;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public class PathConstraintsTest {
  public static final double DELTA = 1e-3;

  @Test
  public void constructor() {
    PathConstraints constraints = new PathConstraints(4.0, 3.0);
    assertEquals(4.0, constraints.maxVelocity, DELTA);
    assertEquals(3.0, constraints.maxAcceleration, DELTA);
  }
}
